src/index.ts

```diff
@@ -3,20 +3,32 @@ import Metrics from './metrics';
 import IStorageProvider from './storage-provider';
 import LocalStorageProvider from './storage-provider-local';
 
+export interface IStaticContext {
+    appName: string;
+    environment?: string;
+}
+
+export interface IMutableContext {
+    userId?: string;
+    sessionId?: string;
+    remoteAddress?: string;
+    properties?: {
+        [key:string]: string,
+    };
+}
+
+export type IContext = IStaticContext & IMutableContext;
+
 export interface IConfig {
+    context: IContext;
```

Comment:
If you make the context optional in the constructor we don't have to do a major version.

Reply:
Ya for this commit, you can't make context optional since this package requires appName. appName is part of the Unleash Context. I still think this commit (e16704a) is the most in-line with Unleash Context, but the newer commits are also pretty clean and don't introduce a change to the api.